### PR TITLE
doc/spec: specify default constratin is not implemented

### PR DESCRIPTION
**Pull request by cuichenli**
Related to https://github.com/cue-lang/cue/issues/1109

### CODE REVIEW

**Comment by myitcv** [comment]
Thanks very much for raising this. 

Please can update the commit message per the [contribution guide](https://github.com/cue-lang/cue/blob/master/doc/contribute.md#good-commit-messages) so that it reads something like:


```
doc/spec: specify default constraint is not implemented

For #1109
```

The first line comment then tells someone briefly what the change corresponds to, the use of `For` in the body relates it to the linked issue but does not close it (this is a convention we have adopted from the Go project).

File: doc/ref/spec.md [hunk 1]
```diff
@@ -1087,14 +1088,15 @@ whose label does not unify with any of the patterns of the pattern
 constraints defined for `a` _and_ for which there exists no field in `a`
 with that label.
 The token `...` is a shorthand for `..._`.
+_Note_: This behavior is not implemented yet.
```
Comment (myitcv) [inline doc/ref/spec.md]:
```
_Note_: default constraints are not yet implemented.
```

File: doc/ref/spec.md [hunk 2]
```diff
@@ -1062,7 +1062,8 @@ Expression                             Result (without optional fields)
 ```
 
 A struct may define constraints that apply to fields that are added when unified
-with another struct using pattern or default constraints.
+with another struct using pattern or default constraints (_Note_: The feature of
```
Comment (myitcv) [inline doc/ref/spec.md]:
```
... default constraints. (_Note_: default constraints are not yet implemented).
```

File: doc/ref/spec.md [hunk 3]
```diff
@@ -1087,14 +1088,15 @@ whose label does not unify with any of the patterns of the pattern
...
+_Note_: This behavior is not implemented yet.
 
 
 ```
 a: {
     foo:    string    // foo is a string
     [=~"^i"]: int     // all other fields starting with i are integers
     [=~"^b"]: bool    // all other fields starting with b are booleans
-    ...string         // all other fields must be a string
+    ...string         // all other fields must be a string. Note: to be implemented.
```
Comment (myitcv) [inline doc/ref/spec.md]:
```
// all other fields must be a string. Note: default constraints are not yet implemented.
```

**Review by myitcv — CHANGES_REQUESTED** [review]
@cuichenli gentle nudge on the commit message format.

Also some proposed changes on the wording to keep things simple and consistent.

Otherwise, we're very happy to add these notes to the spec, not least because it's the reference many people look towards.

**Comment by cuichenli** [comment]
@myitcv sorry i was away for a while and just got back. 
just updated the commit message and wording. please kindly check again. thanks

**Review by myitcv — APPROVED** [review]
LGTM. Thanks very much for your patience fixing up commits messages etc.

**Comment by myitcv** [comment]
Submitted as 1c52d6a7d693133f04810da355f8036848c2c78a

**Pull request closed without merging**